src/Telegram.Bot/Requests/Available methods/Manage Chat/RestrictChatMemberRequest.cs

```diff
@@ -1,68 +1,51 @@
+    /// </summary>
     [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
-    public required ChatId ChatId { get; init; }
+    public required ChatId ChatId { get; set; }
 
-    /// <inheritdoc />
-    [JsonRequired]
+    /// <summary>
+    /// Unique identifier of the target user
+    /// </summary>
     [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
-    public required long UserId { get; init; }
+    public required long UserId { get; set; }
 
     /// <summary>
-    /// New user permissions
+    /// An object for new user permissions
     /// </summary>
-    [JsonRequired]
     [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
-    public required ChatPermissions Permissions { get; init; }
+    public required ChatPermissions Permissions { get; set; }
 
     /// <summary>
-    /// Pass <see langword="true"/> if chat permissions are set independently. Otherwise, the
-    /// <see cref="ChatPermissions.CanSendOtherMessages"/>, and <see cref="ChatPermissions.CanAddWebPagePreviews"/>
-    /// permissions will imply the <see cref="ChatPermissions.CanSendMessages"/>,
-    /// <see cref="ChatPermissions.CanSendAudios"/>, <see cref="ChatPermissions.CanSendDocuments"/>,
-    /// <see cref="ChatPermissions.CanSendPhotos"/>, <see cref="ChatPermissions.CanSendVideos"/>,
-    /// <see cref="ChatPermissions.CanSendVideoNotes"/>, and <see cref="ChatPermissions.CanSendVoiceNotes"/>
-    /// permissions; the <see cref="ChatPermissions.CanSendPolls"/> permission will imply the
-    /// <see cref="ChatPermissions.CanSendMessages"/> permission.
+    /// Pass <see langword="true"/> if chat permissions are set independently. Otherwise, the <em>CanSendOtherMessages</em> and <em>CanAddWebPagePreviews</em> permissions will imply the <em>CanSendMessages</em>, <em>CanSendAudios</em>, <em>CanSendDocuments</em>, <em>CanSendPhotos</em>, <em>CanSendVideos</em>, <em>CanSendVideoNotes</em>, and <em>CanSendVoiceNotes</em> permissions; the <em>CanSendPolls</em> permission will imply the <em>CanSendMessages</em> permission.
     /// </summary>
     [JsonInclude]
-    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    public bool? UseIndependentChatPermissions { get; set; }
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
+    public bool UseIndependentChatPermissions { get; set; }
 
     /// <summary>
-    /// Date when restrictions will be lifted for the user, unix time. If user is restricted for
-    /// more than 366 days or less than 30 seconds from the current time, they are considered to
-    /// be restricted forever.
+    /// Date when restrictions will be lifted for the user, in UTC. If user is restricted for more than 366 days or less than 30 seconds from the current time, they are considered to be restricted forever
     /// </summary>
-    [JsonConverter(typeof(UnixDateTimeConverter))]
     [JsonInclude]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [JsonConverter(typeof(BanTimeConverter))]
```

Comment:
Why was the converter changes? BanTimeConverter was written for a different time if I remember it correctly

Reply:
generator of code sees "DateTime" "until_date" and considers it's a ban time.

now that BanTimeConverter just produce a 0 when DateTime is unspecified, which is fine for this method (means forever)

>Date when restrictions will be lifted for the user; Unix time. If user is restricted for more than 366 days or less than 30 seconds from the current time, they are considered to be restricted forever